python/lsst/daf/butler/core/registry.py

```diff
@@ -202,6 +253,15 @@ def __init__(self, registryConfig, schemaConfig=None, dimensionConfig=None, crea
     def __str__(self):
         return "None"
 
+    @property
+    def connectionString(self):
+        """Return the connection string to the underlying database
+        (`sqlalchemy.engine.url.URL`).
+        """
+        # this import can not live at the top due to circular import issue
```

Comment:
I guess moving `RegistryConfig` into its own module would break this circular import?  You might consider that.

Reply:
I don't think htat's the case. The problem is that I'd like for `ConnectionStringFactory` to "downgrade/upgrade" any type of config to a `RegistryConfig`. That's why I'm recasting any config or string sent to `ConnectionStringFactory.fromConfig` into a `RegistryConfig`. 

Meanwhile I am using the `ConnectionStringFactory` in the `RegistryConfig`s methods. This wouldn't be the case if the methods are left in the `Registry`, but since we moved them to `RegistryConfig` the problem is solvable only if I make `ConnectionStringFactory.fromConfig` into a `ConnectionStringFactory.fromRegistryConfig` but I think that's very impractical for the cost of a single import line in a single classmethod. 

I went forward and separated the `RegistryConfig` into a new module anyhow because there's some "meat" on it now so I thought it looked neater. I can bring it back though if you don't think it's big enough for that or if It would just be better if they stay bundled like `Datastore` and `DatastoreConfig` (question for @timj as well).

Reply:
Separate is fine now that there's real code in there.